pkg/egressgateway/manager.go

```diff
@@ -516,33 +512,30 @@ func (manager *Manager) handleNodeEvent(event resource.Event[*cilium_api_v2.Cili
 	manager.Lock()
 	defer manager.Unlock()
 
-	if event.Kind == resource.Upsert {
-		manager.nodeDataStore[node.Name] = node
-		manager.onChangeNodeLocked(eventUpdateNode)
-	} else {
-		delete(manager.nodeDataStore, node.Name)
-		manager.onChangeNodeLocked(eventDeleteNode)
-	}
-}
+	// Find the node if we already have it.
+	nidx, found := slices.BinarySearchFunc(manager.nodes, node, func(a nodeTypes.Node, b nodeTypes.Node) int {
+		return cmp.Compare(a.Name, b.Name)
+	})
 
-func (manager *Manager) onChangeNodeLocked(e eventType) {
-	manager.nodes = []nodeTypes.Node{}
-	for _, n := range manager.nodeDataStore {
-		manager.nodes = append(manager.nodes, n)
+	if event.Kind == resource.Delete {
+		// Delete the node if we're aware of it.
+		if found {
+			manager.nodes = slices.Delete(manager.nodes, nidx, nidx+1)
+		}
+
+		manager.reconciliationTrigger.TriggerWithReason("node deleted")
+		return
 	}
-	sort.Slice(manager.nodes, func(i, j int) bool {
-		return manager.nodes[i].Name < manager.nodes[j].Name
-	})
 
-	reason := ""
-	if e == eventUpdateNode {
-		reason = "node updated"
-	} else if e == eventDeleteNode {
-		reason = "node deleted"
+	// Update the node if we have it, otherwise insert in the correct
+	// position.
+	if found {
+		manager.nodes[nidx] = node
+	} else {
+		manager.nodes = slices.Insert(manager.nodes, nidx, node)
 	}
 
-	manager.setEventBitmap(e)
```

Comment:
No more setting the event bitmap?

Reply:
I don't see those events being used anywhere so I figured it's obsolete.

Reply:
cc @jibi

Reply:
if they aren't used we should get rid of the method as well.

Reply:
The event bitmap is used but the node update events are never read from the bitmap. So to me it made little sense to set the event bitmap in the case of node events. I think the method itself is still widely used for other event types.